src/cli/argument-parser.js

```diff
@@ -107,6 +107,7 @@ export default class CLIArgumentParser {
             .option('--ports <port1,port2>', 'specify custom port numbers')
             .option('--hostname <name>', 'specify the hostname')
             .option('--proxy <host>', 'specify the host of the proxy server')
+            .option('--ssl', 'run the TestCafe proxy server over the HTTPS protocol')
```

Comment:
```js
            .option('--ssl', 'run TestCafe proxy server over the HTTPS protocol')
```

Maybe we should rename the flag to `--https` considering the description?

Reply:
No. Need to change description.
This flag contains ssl options that will passed for https server creation.

Also, this option has same name (`ssl`) for the widely known [http-server](https://www.npmjs.com/package/http-server) module.
Changed description: `specify ssl options for starting TestCafe using https protocol`

Reply:
> `specify ssl options`

So, you can actually specify some options after this flag? Which options?

Reply:
Yes, you ca specify a lot of options (all options from [https server options](https://nodejs.org/api/https.html#https_https_createserver_options_requestlistener) and parent classes: [http server iptions](https://nodejs.org/api/tls.html#tls_tls_createserver_options_secureconnectionlistener), [tls server options](https://nodejs.org/api/tls.html#tls_tls_createsecurecontext_options)).

See simple example in https://github.com/devexpress/testcafe/issues/1985#issuecomment-389488983

Reply:
Cool!
As far as I understand, options are not required, so we can write

```js
            .option('--ssl <options>', 'enables access to TestCafe via the HTTPS protocol and allows you to specify SSL options')
```

Reply:
`specify SSL options for starting TestCafe proxy server over the HTTPS protocol`

Reply:
```js
            .option('--ssl <options>', 'specify SSL options to run TestCafe proxy server over the HTTPS protocol');
```